### ISSUE TITLE
Size predicate validator

### ISSUE THREAD

**Issue by DenisNovac** [fthomas/refined#1155]
There is no validator for types refined with `Size[n]` predicate. Which leads to an error:

> could not find implicit value for parameter v: eu.timepit.refined.api.Validate[Long, T]

In "Practical FP in Scala" book there is this workaround (which didn't worked for me actually):

```scala
import eu.timepit.refined.api.Validate
import eu.timepit.refined.collection.Size

trait RefinedValidateSize {

  implicit def validateSizeN[N <: Int, R](implicit
      w: ValueOf[N]
  ): Validate.Plain[R, Size[N]] =
    Validate.fromPredicate[R, Size[N]](
      _.toString.length == w.value,
      _ => s"Must have ${w.value} digits",
      Size[N](w.value)
    )

}
```

But is there any reason there is no such thing in library?

**Comment by fthomas** [comment]
Try `Size[Equal[n]]`.

**Comment by DenisNovac** [comment]
Validator is still missing.

I also tried to rewrite self-written validator with `Equal` but the macros error is still here.

```scala
  implicit def validateSizeN[N <: Int, R](implicit
                                          w: ValueOf[N]
                                         ): Validate.Plain[R, Size[Equal[N]]] =
    Validate.fromPredicate[R, Size[Equal[N]]](
      _.toString.length == w.value,
      _ => s"Must have ${w.value} digits",
      Size[Equal[N]](Equal[N](w.value))
    )
```

**Comment by fthomas** [comment]
> Validator is still missing.

Right. You're trying to check if the string representation of an `Int` has `n` digits. We don't have a predicate for that.

> I also tried to rewrite self-written validator with `Equal` but the macros error is still here.

Looks like https://github.com/fthomas/refined/blob/master/modules/docs/macro_pitfalls.md. Custom predicates + macros probably won't work in Scastie. `refineV` works: https://scastie.scala-lang.org/3nGUIQ9tSdKGmQ2lNZE8fw

**Comment by DenisNovac** [comment]
They are not working in the IDEA either. Tried to write exact types instead type aliases and it helped. Thanks for helping out!

I didn't really got what it is about in macro_pitfalls. I should replace my `SixP` alias and validate instance to a different project so i could use `refineV` directly on it instead of `Int Refined Size[6]`?

**Comment by fthomas** [comment]
> I didn't really got what it is about in macro_pitfalls.

My recommendation is to not use refined's macros at all. They cause problems like above and are not available on Scala 3.

> I should replace my `SixP` alias and validate instance to a different project so i could use `refineV` directly on it instead of `Int Refined Size[6]`?

No, the alias is fine. If you don't use macros, there is also no need to move the `Validate` instance to another project. 

What I've been doing lately is to define a type alias + an object with the same name that extends `RefinedTypeOps` like this:
```scala
import eu.timepit.refined.api.RefinedTypeOps

type SixP = Int Refined Size[6]
object SixP extends RefinedTypeOps[Int Refined Size[6], Int]
```
which gives you nice syntax for creating `SixP` instances, like `SixP.from(123456)`. Here is a Scastie that demonstrates this: https://scastie.scala-lang.org/njwd5esnRpe6daeNizS0HQ

**Comment by DenisNovac** [comment]
Got it, thanks for clarifying!
Is there any chance of having `Validate` for `Size` predicate in refined out of the box?
